### ISSUE TITLE
FFT execution failure in ROCm 4.2.0

### ISSUE THREAD

**Issue by takagi** [cupy/cupy#5314]
I'm investigating a number of testing failures of `cupy.fft` under ROCm 4.2.0. Some fft execution fails with return code `CUFFT_EXEC_FAILED`, that is actually `HIPFFT_EXEC_FAILED` in hipFFT, and it seems that rocFFT internally called returns `rocfft_status_invalid_work_buffer` there.

```
________________________ TestPlanCache.test_LRU_cache4 _________________________

self = <cupy_tests.fft_tests.test_cache.TestPlanCache testMethod=test_LRU_cache4>

    def test_LRU_cache4(self):
        # test if fetching the plan will reorder it to the top
        cache = config.get_plan_cache()
        assert cache.get_curr_size() == 0 <= cache.get_size()

        # this creates a Plan1d
        a = testing.shaped_random((10,), cupy, cupy.float32)
        cupy.fft.fft(a)
        assert cache.get_curr_size() == 1 <= cache.get_size()

        # this creates a PlanNd
        a = testing.shaped_random((10, 20), cupy, cupy.float32)
>       cupy.fft.fftn(a)

tests/cupy_tests/fft_tests/test_cache.py:120:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cupy/fft/_fft.py:772: in fftn
    return func(a, s, axes, norm, cufft.CUFFT_FORWARD)
cupy/fft/_fft.py:614: in _fftn
    a = _exec_fftn(a, direction, value_type, norm=norm, axes=axes_sorted,
cupy/fft/_fft.py:554: in _exec_fftn
    plan.fft(a, out, direction)
cupy/cuda/cufft.pyx:838: in cupy.cuda.cufft.PlanNd.fft
    execC2C(plan, a.data.ptr, out.data.ptr, direction)
cupy/cuda/cufft.pyx:1100: in cupy.cuda.cufft.execC2C
    check_result(result)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   raise CuFFTError(result)
E   cupy.cuda.cufft.CuFFTError: CUFFT_EXEC_FAILED

cupy/cuda/cufft.pyx:169: CuFFTError
=========================== short test summary info ============================
FAILED tests/cupy_tests/fft_tests/test_cache.py::TestPlanCache::test_LRU_cache4
```

**Comment by leofang** [comment]
Is this with #5311 fixed?

**Comment by takagi** [comment]
No, this is another problem.

**Comment by takagi** [comment]
It seems that hipFFT included rocFFT is deprecated since ROCm 4.1, and we need to one from hipFFT repository.
https://github.com/ROCmSoftwarePlatform/hipFFT

**Comment by kmaehashi** [comment]
Fixed via https://github.com/cupy/cupy/pull/5318
